pkg/controller/cluster/main-controller.go

```diff
@@ -233,6 +238,7 @@ func NewController(
 	tenantInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: controller.enqueueTenant,
 		UpdateFunc: func(old, new interface{}) {
+			// TODO: compare old vs new and don't enqueue if they are identical
```

Comment:
Can you not use reflect.Equal(old, new) ?

Reply:
There's a package from kubernetes for that `equality.Semantic.DeepEqual` and `equality.Semantic.DeepDerivative` I experimented with both, they didn't work quite as I expected, I'll keep looking into it.